Library/Homebrew/formula_installer.rb

```diff
@@ -572,6 +572,7 @@ def inherited_options_for(dep)
   end
 
   def install_dependencies(deps)
+    return if formula.full_name == "patchelf" unless OS.mac?
```

Comment:
I'd suggest checking if `formula.full_name` equals to `dep.to_formula.full_name` (on line 581 below) and skipping that dep in such a case.... and submitting this upstream.

Reply:
This would solve circular dependencies once and for all 🤺

Reply:
Yes, why not. Not sure this is the best solution, because I actually not fully understand how the dependency cycle is happening. Maybe patchelf should not be added as a patchelf dependency at a first place. So not sure we are fixing this at the right place. But that should at least get us started.

Reply:
I'll keep it in mind, I want 1.6.15 out as soon possible to be able to carry on with the core merges, and preparer 1.7.0.